legacy-commands-v1/src/main/java/io/github/minecraftcursedlegacy/api/command/DispatcherRegistry.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2020 The Cursed Legacy Team.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.minecraftcursedlegacy.api.command;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.entity.player.Player;
+
+/**
+ * Registry for for mapping commands to the relevant {@linkplain CommandDispatcher command dispatchers}.
+ * @since 1.1.0
+ */
+public class DispatcherRegistry {
+	private static final Map<String, CommandDispatcher> SINGLEPLAYER_COMMANDS = new HashMap<>();
+	private static final Map<String, CommandDispatcher> MULTIPLAYER_COMMANDS = new HashMap<>();
+
+	/**
+	 * Set the dispatcher associated with this command on singleplayer.
+	 * @param command the command to set the dispatcher for.
+	 * @param dispatcher the dispatcher in charge of handling and dispatching this command.
+	 */
+	public static void registerSingleplayerDispatcher(String command, CommandDispatcher dispatcher) {
```

Comment:
I really don't like the idea of registering a dispatcher per-command-name. It's conceivable that some handlers might not know ahead of time.

Reply:
How would you handle conflicts with multiple dispatchers handling the same command name if the dispatchers don't know the commands they're meant to be handling?

Reply:
Either let them all handle it, or let the first one to be registered handle it. In #35 I did the latter. IMO it's a bit of a non-issue as the chances of the same command being registered to two different dispatchers is low. (I think it's unlikely that multiple different dispatchers are likely to be used at all, tbh)

Reply:
what about B O T H

Reply:
yeef